### ISSUE TITLE
Missing Syntax Grammar Reference from In This Section

### ISSUE THREAD

**Issue by Player1UK** [MicrosoftDocs/cpp-docs#1123]
Directives Reference, Symbols Reference, and Operators Reference all mention and contain syntax grammar instances without linking to the essential Syntax Grammar Reference.

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1d47dd35-d319-548d-db9f-6e1d754a4dc6
* Version Independent ID: f97c5b9d-b51c-860d-b644-11fb50f43884
* Content: [Microsoft Macro Assembler Reference](https://docs.microsoft.com/en-gb/cpp/assembler/masm/microsoft-macro-assembler-reference?view=vs-2019)
* Content Source: [docs/assembler/masm/microsoft-macro-assembler-reference.md](https://github.com/Microsoft/cpp-docs/blob/master/docs/assembler/masm/microsoft-macro-assembler-reference.md)
* Product: **visual-cpp**
* Technology: **cpp-masm**
* GitHub Login: @corob-msft
* Microsoft Alias: **corob**

**Comment by mikeblome** [comment]
@Player1UK can you pls clarify what you think needs to happen here? Thanks.

**Comment by Player1UK** [comment]
A whole reference section on Syntax Grammar, much like Appendix B from the original MASM 6.1 manual, yet updated for modern 64-bit assembly language.

Please see attached document for all necessary grammar entries.  Updated to use previously mentioned Microsoft style of single brackets over double brackets.

With this section, keyword syntax entries will be able to link directly to the essential syntax grammar entry to further complete understanding.
[64-bit MASM Syntax - BNF Grammar.docx](https://github.com/MicrosoftDocs/cpp-docs/files/3876565/64-bit.MASM.Syntax.-.BNF.Grammar.docx)

**Comment by mikeblome** [comment]
@player1UK thanks very much for your feedback on the MASM content. You identified many instances where the content was out of date. We have addressed all those instances where we agreed that a fix was needed. However, it is not a goal to have reference content be "BNF complete". Our policy is to use descriptive parameter names and to include all necessary information about allowed values, etc. as remarks. That being said, the team did agree that having a pseudo-BNF grammar for 64-bit MASM might serve some limited purposes and so we have added that topic and we link to it from the reference topics. It is not a goal to have that topic be a formal language specification. For example, you will find parameter names in the reference topics that are tailored to the specific directive and do not have an entry in the BNF grammar as a nonterminal. If you find significant errors or omissions that hinder a programmer's ability to understand and use MASM, you are of course welcome to file an issue, or even better, a pull request.

https://docs.microsoft.com/en-us/cpp/assembler/masm/masm-bnf-grammar?view=vs-2019
